// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;

option java_outer_classname = "TimeSyncServiceProto";

import "bosdyn/api/time_sync.proto";

// The time-sync service estimates the difference between server and client clocks.
// Time synchronization is a tool which allows applications to work in a unified timebase with
// precision. It is useful in cases where a precise time must be set, independently of network
// communication lag. In distributed systems and robotics, hardware, system-level, and per-process
// approaches can be used to obtain synchronization.
// This service implements a stand alone time synchronization service. It enables clients to
// establish a per-process offset between two processes which may be on separate systems.
service TimeSyncService {
    // See the exchange documentation in time_sync.proto. This call makes one client/server
    // round trip toward clock synchronization.
    rpc TimeSyncUpdate(TimeSyncUpdateRequest) returns (TimeSyncUpdateResponse) {}

}


